grafana/resource_datasource_permission.go

```diff
@@ -119,15 +129,21 @@ func ReadDatasourcePermissions(ctx context.Context, d *schema.ResourceData, meta
 	}
 
 	permissionItems := make([]interface{}, len(response.Permissions))
-	for i, permission := range response.Permissions {
+	for _, permission := range response.Permissions {
```

Comment:
Note that the line above creates a non-empty slice of this type, meaning that in line 146, when you append, you're appending to the slice and thus will end up with `2 * len(response.Permissions)`, see [this example](https://go.dev/play/p/6BN7o9WGyUm).

What you should do is keep `i` and assign to `permissionItems[i]` as it was done previously.

Reply:
Ouph, that was a typo, I forgot to specify both length and capacity for the slice. I've fixed it now, can you take another look at it please?

Reply:
Cool. Let me know when ready to be reviewed again.